h2integrate/control/control_strategies/pyomo_controllers.py

```diff
@@ -0,0 +1,757 @@
+            between 0 and 1 (e.g., 0.9 for 90% efficiency).
+        discharge_efficiency (float): Efficiency of discharging the storage, represented as a
+            decimal between 0 and 1 (e.g., 0.9 for 90% efficiency).
+    """
+
+    max_charge_rate: float = field()
+    max_discharge_rate: float = field()
+    charge_efficiency: float = field()
+    discharge_efficiency: float = field()
+
+
+class PyomoControllerH2Storage(PyomoControllerBaseClass):
+    def setup(self):
+        self.config = PyomoControllerH2StorageConfig.from_dict(
+            merge_shared_inputs(self.options["tech_config"]["model_inputs"], "control")
+        )
+        super().setup()
+
+
+class SimpleBatteryControllerHeuristic(PyomoControllerBaseClass):
+    """Fixes battery dispatch operations based on user input.
+
+    Currently, enforces available generation and grid limit assuming no battery charging from grid.
+
+    """
+
+    def setup(self):
+        """Initialize SimpleBatteryControllerHeuristic.
+
+        Args:
+            pyomo_model (pyomo.ConcreteModel): Pyomo concrete model.
+            index_set (pyomo.Set): Indexed set.
+            system_model (PySAMBatteryModel.BatteryStateful): Battery system model.
+            fixed_dispatch (Optional[List], optional): List of normalized values [-1, 1]
+                (Charging (-), Discharging (+)). Defaults to None.
+            block_set_name (str, optional): Name of block set. Defaults to 'heuristic_battery'.
+            control_options (dict, optional): Dispatch options. Defaults to None.
+
+        """
+        super().setup()
+
+        # self._create_soc_linking_constraint()
+
+        # TODO: implement and test lifecycle counting
+        # TODO: we could remove this option and just have lifecycle count default
+        # self.control_options = control_options
+        # if self.control_options.include_lifecycle_count:
+        #     self._create_lifecycle_model()
+        #     if self.control_options.max_lifecycle_per_day < np.inf:
+        #         self._create_lifecycle_count_constraint()
+
+        self.round_digits = 4
+
+        self.max_charge_fraction = [0.0] * self.config.n_control_window
+        self.max_discharge_fraction = [0.0] * self.config.n_control_window
+        self._fixed_dispatch = [0.0] * self.config.n_control_window
+
+        # TODO: should I enforce either a day schedule or a year schedule year and save it as
+        # user input? Additionally, Should I drop it as input in the init function?
+        # if fixed_dispatch is not None:
+        #     self.user_fixed_dispatch = fixed_dispatch
+
+    def initialize_parameters(self):
+        """Initializes parameters."""
+        # TODO: implement and test lifecycle counting
+        # if self.config.include_lifecycle_count:
+        #     self.lifecycle_cost = (
+        #         self.options.lifecycle_cost_per_kWh_cycle
+        #         * self._system_model.value("nominal_energy")
+        #     )
```

Comment:
There are a few TODOs in this file and it's unclear to me if they're intended to be done as part of this PR, done at some future point (if so, please make an issue for them), or if they're just holdovers from HOPP in some way and may or may not be relevant anymore. Could you please clarify?

Reply:
@jaredthomas68